lib/samples/analyze_network_with_subnetwork_trace/analyze_network_with_subnetwork_trace.dart

```diff
@@ -0,0 +1,691 @@
+      _configuration!.traversability?.barriers = _initialBarrierExpression;
+    }
+
+    // Clear the condition form.
+    _clearForm();
+  }
+
+  // A human-readable label for each utility attribute comparison operator.
+  String _operatorLabel(UtilityAttributeComparisonOperator op) {
+    switch (op) {
+      case UtilityAttributeComparisonOperator.equal:
+        return 'Equal';
+      case UtilityAttributeComparisonOperator.notEqual:
+        return 'Not Equal';
+      case UtilityAttributeComparisonOperator.greaterThan:
+        return 'Greater Than';
+      case UtilityAttributeComparisonOperator.greaterThanEqual:
+        return 'Greater Than Equal';
+      case UtilityAttributeComparisonOperator.lessThan:
+        return 'Less Than';
+      case UtilityAttributeComparisonOperator.lessThanEqual:
+        return 'Less Than Equal';
+      case UtilityAttributeComparisonOperator.includesTheValues:
+        return 'Includes The Values';
+      case UtilityAttributeComparisonOperator.doesNotIncludeTheValues:
+        return 'Does Not Include The Values';
+      case UtilityAttributeComparisonOperator.includesAny:
+        return 'Includes Any';
+      case UtilityAttributeComparisonOperator.doesNotIncludeAny:
+        return 'Does Not Include Any';
+    }
+  }
+
+  // Converts the values to matching data types.
+  dynamic _convertToDataType(
+    String raw,
+    UtilityNetworkAttributeDataType dataType,
+  ) {
+    switch (dataType) {
+      case UtilityNetworkAttributeDataType.integer:
+        return int.tryParse(raw) ?? raw;
+      case UtilityNetworkAttributeDataType.float:
+      case UtilityNetworkAttributeDataType.double:
+        return double.tryParse(raw) ?? raw;
+      case UtilityNetworkAttributeDataType.boolean:
+        return raw.toLowerCase() == 'true';
+    }
+  }
+
+  // Gets the string representation of a condition for display.
+  String _conditionString(UtilityTraceConditionalExpression condition) {
+    if (condition is UtilityNetworkAttributeComparison) {
+      final attributeName = condition.networkAttribute.name;
+      final operatorLabel = _operatorLabel(condition.comparisonOperator);
+      final value = condition.value;
+
+      if (condition.networkAttribute.domain is CodedValueDomain) {
+        final domain = condition.networkAttribute.domain! as CodedValueDomain;
+        // Try to find the coded value that matches the condition value
+        final codedValue = domain.codedValues.firstWhere(
+          (cv) => cv.code == value,
+        );
+        return "'$attributeName' $operatorLabel '${codedValue.name}'";
+      } else {
+        return "'$attributeName' $operatorLabel '$value'";
+      }
+    }
+    return condition.toString();
+  }
+
+  // Clears the condition form inputs.
+  void _clearForm() {
+    _selectedAttribute = null;
+    _selectedValue = null;
+    _valueController.clear();
+    _codedValues = [];
+
+    // Refresh the UI.
+    setState(() {});
```

Comment:
Instead of using `setState(() {})`, the relevant state changes should be inside the callback:
```dart
  _valueController.clear();
  setState(() {
    _selectedAttribute = null;
    _selectedValue = null;
    _codedValues = [];
  });
```